assets/js-image-loading.js

```diff
@@ -19,124 +21,329 @@ const handleImageLoading = () => {
     }
   }
 
-  const wrappers = document.querySelectorAll(`.${imageOptions.classes.wrapper}`);
+  const wrappers = document.querySelectorAll(`.${config.classes.wrapper}`),
+        mobileWidth = 992; // Define mobile width for responsive design
+
   if (!wrappers.length) return false;
 
-  const isInViewport = (el) => {
-    const rect = el.getBoundingClientRect();
-    return (
-      rect.top >= 0 &&
-      rect.left >= 0 &&
-      rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
-      rect.right <= (window.innerWidth || document.documentElement.clientWidth)
-    )
+  let observer = null;
+
+  // Helper function to decode lazy-loaded images
+  // Only used for images that are lazy-loaded but become visible
+  const decodeImage = (image) => {
+    if (!('decode' in image)) return;
+
+    image.decode().catch(() => {
+      // Silently fail - the image will still display normally
+    })
   }
 
   const sourcesDataLoad = (mainImage) => {
-    const wrapper = mainImage.closest(`.${imageOptions.classes.wrapper}`),
-          sources = wrapper.querySelectorAll(`source[${imageOptions.attributes.sourceSrcset}]`);
+    const wrapper = mainImage.closest(`.${config.classes.wrapper}`),
+          sources = wrapper.querySelectorAll(`source[${config.attributes.sourceSrcset}]`),
+          inViewport = $.is($.inViewport, 'function') ? $.inViewport(mainImage) : false,
+          slowConnection = $.slowConnection();
 
     if (sources.length) {
       sources.forEach(source => {
-        const dataSrc = source.getAttribute(`${imageOptions.attributes.sourceSrcset}`);
+        const dataSrc = source.getAttribute(config.attributes.sourceSrcset);
 
         if (dataSrc) {
           source.setAttribute('srcset', dataSrc);
-          source.removeAttribute(`${imageOptions.attributes.sourceSrcset}`);
+          source.removeAttribute(config.attributes.sourceSrcset);
+
+          // Use Priority Hints if available (experimental)
+          if ('importance' in source && !slowConnection && inViewport) {
```

Comment:
I couldn't find any docs for importance? Is still being used?

I saw about it here https://web.dev/articles/fetch-priority

> Fetch Priority was first experimented with in Chrome as an origin trial in 2018, and then again in 2021 using the importance attribute. At that time it was called [Priority Hints](https://github.com/WICG/priority-hints). The interface has since changed to fetchpriority for HTML and priority for JavaScript's Fetch API as part of the web standards process. To reduce confusion, we now call this API Fetch Priority.

So not sure if we should support it?

Reply:
Got it 👍